src/hmm/hmm-utils.cc

```diff
@@ -94,10 +94,15 @@ fst::VectorFst<fst::StdArc> *GetHmmAsFst(
        hmm_state < static_cast<int32>(entry.size());
        hmm_state++) {
     int32 pdf_class = entry[hmm_state].pdf_class, pdf;
-    if (pdf_class == kNoPdf) pdf = kNoPdf;  // nonemitting state.
-    else {
+    int32 self_loop_pdf_class = entry[hmm_state].pdf_class, self_loop_pdf;
```

Comment:
OK, then it's a bug-- I think you meant
int32 self_loop_pdf_class = entry[hmm_state].**self_loop**_pdf_class